doc/source/examples/extended_examples/static_rom/static_rom_data_generation.py

```diff
@@ -0,0 +1,302 @@
+        take_mid_nodes=True,
+    )
+    # Get output data
+    connected_nodes_scoping = op.outputs.mesh_scoping()
+    # Compress the list to only keep unique IDs
+    connected_nodes_scoping.ids = sorted(list(set(connected_nodes_scoping.ids)))
+    return connected_nodes_scoping
+
+
+def write_points(model: dpf.Model, scoping: dpf.Scoping, output_folder: str | Path):
+    """Write points.bin file."""
+    nodes = model.metadata.meshed_region.nodes
+    scoped_node_indices, _ = nodes.map_scoping(scoping)
+    points_coordinates = nodes.coordinates_field.data[scoped_node_indices]
+    write_binary(Path(output_folder).joinpath("points.bin"), points_coordinates)
+
+
+def write_doe_headers(output_folder: str | Path, name: str, parameters: dict):
+    """Write blank doe.csv file with headers."""
+    with open(Path(output_folder).joinpath("doe.csv"), "w", newline="") as fw:
+        writer = csv.writer(fw)
+        writer.writerow([name] + list(parameters.keys()))
+
+
+def write_doe_entry(output_folder: str | Path, snapshot_name: str, parameters: dict):
+    """Write entry to doe.csv file."""
+    with open(Path(output_folder).joinpath("doe.csv"), "a", newline="") as fw:
+        writer = csv.writer(fw)
+        writer.writerow([snapshot_name] + list(parameters.values()))
+
+
+def export_static_ROM_variation(
+    model: dpf.Model,
+    scoping: dpf.Scoping,
+    name: str,
+    output_folder: str | Path,
+    parameters: dict,
+    snap_idx: int = 0,
+    new_metadata: bool = False,
+):
+    """
+    Export static ROM data for one parameter variation.
+
+    Parameters
+    ----------
+    model : dpf.Model
+        DPF model with results data loaded.
+    scoping : dpf.Scoping
+        DPF nodal scoping for result export.
+    name : str
+        result quantity to export. Valid options are `displacement` and `stress`.
+    output_folder : str|Path
+        exported data will be stored in this folder. Use separate folders for each physics type.
+    parameters : dict
+        dictionary of name-value pairs for the input parameters used to generate the current
+        results.
+    snap_idx : int, default = 0
+        unique ID for the current results.
+    new_metadata : bool, default = False
+        used the first time the function is called for a given data generation run. Triggers the
+        creation of ``points.bin``, ``settings.json`` and a new ``doe.csv`` file. Existing files are
+        overrwritten.
+    """
+    # Create the output folder
+    output_folder = Path(output_folder)
+    output_folder.mkdir(parents=True, exist_ok=True)
+
+    # Modify this section to export additional result types
+    is_deformation = False
+    if name == "displacement":
+        result = model.results.displacement
+        is_deformation = True
+    elif name == "stress":
+        result = model.results.stress
+    else:
+        raise ValueError(f"Unsupported result type: {name}")
+
+    # Retrieve displacement and stress at last result set.
```

Comment:
```suggestion
    # Retrieve displacement and stress at last result set
```
Consistently either use or not use concluding punctuation in code comments. The comments here do not tend to use them.

Reply:
Done, thanks.